packages/core/src/apiProviders/jsonRpc.ts

```diff
@@ -0,0 +1,31 @@
+import { providers } from 'ethers'
+
+import { Chain } from '../types'
+
+import { ApiProvider } from './ApiProvider'
+
+export const jsonRpcProvider = (
+  getRpcUrls: (chain: Chain) => { rpcUrl: string; webSocketRpcUrl?: string },
+): ApiProvider<
+  providers.StaticJsonRpcProvider,
+  providers.WebSocketProvider
+> => {
+  return function (chain) {
+    const { rpcUrl, webSocketRpcUrl } = getRpcUrls(chain)
+    if (rpcUrl === '') return null
+    return {
+      chain: {
+        ...chain,
+        rpcUrls: {
+          ...chain.rpcUrls,
+          default: rpcUrl,
+        },
+      },
+      provider: () => new providers.StaticJsonRpcProvider(rpcUrl, chain.id),
```

Comment:
I'm all for using `StaticJsonRpcProvider` (cuts down on unnecessary calls), but it might be confusing that `jsonRpcProvider` doesn't use [`JsonRpcProvider`](https://docs.ethers.io/v5/api/providers/jsonrpc-provider/#JsonRpcProvider). Could add a `static` param that is defaulted to `true` or something.

Reply:
maybe i'll rename this to `staticJsonRpcProvider`, might make types easier.